Modules/nismodule.c

```diff
@@ -44,12 +44,47 @@ PyDoc_STRVAR(maps__doc__,
 Returns an array of all available NIS maps within a domain. If domain\n\
 is not specified it defaults to the system default domain.\n");
 
-static PyObject *NisError;
+typedef struct {
+    PyObject *nis_error;
+} nis_state;
+
+static inline nis_state*
+get_nis_state(PyObject *module)
+{
+    void *state = PyModule_GetState(module);
+    assert(state != NULL);
+    return (nis_state *)state;
+}
+
+static int
+nis_clear(PyObject *m)
+{
+    Py_CLEAR(get_nis_state(m)->nis_error);
+    return 0;
+}
+
+static int
+nis_traverse(PyObject *m, visitproc visit, void *arg)
+{
+    Py_VISIT(get_nis_state(m)->nis_error);
+    return 0;
+}
+
+static void
+nis_free(void *m)
+{
+    nis_clear((PyObject *) m);
+}
+
+static struct PyModuleDef nismodule;
+
+#define nis_state_global \
+((nis_state *) PyModule_GetState(PyState_FindModule(&nismodule)))
```

Comment:
This goes against PEP 489. Why not passing the module or the state to nis_error()?

Reply:
I'd like to apply it with the PR related to PEP 489 :)